File: frontend/src/types/itinerary.ts
export interface Itinerary {
  id: string;
  user_id: string;
  title: string;
  description?: string;
  destination: string;
  start_date?: string;
  end_date?: string;
  days: number;
  budget?: 'low' | 'medium' | 'high';
  travel_style?: string;
  status: 'draft' | 'published' | 'archived';
  data: ItineraryData;
  created_at: string;
  updated_at: string;
}

export interface ItineraryData {
  days: DayItinerary[];
  summary?: string;
  total_budget?: number;
}

export interface DayItinerary {
  day: number;
  date?: string;
  activities: Activity[];
}

export interface Activity {
  time: string;
  place: Place;
  description: string;
  duration?: number;
}

export interface Place {
  place_id: string;
  name: string;
  address: string;
  location: {
    lat: number;
    lng: number;
  };
  rating?: number;
  photos?: string[];
  types?: string[];
}

File: backend/app/api/v1/schemas/itinerary.py
from pydantic import BaseModel
from typing import Optional, Dict, Any
from datetime import date, datetime


class ItineraryCreate(BaseModel):
    title: str
    description: Optional[str] = None
    destination: str
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    days: int
    budget: Optional[str] = None
    travel_style: Optional[str] = None
    travelers: Optional[int] = 1
    data: Dict[str, Any]


class ItineraryResponse(BaseModel):
    id: str
    user_id: str
    title: str
    description: Optional[str] = None
    destination: str
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    days: int
    budget: Optional[str] = None
    travel_style: Optional[str] = None
    status: str
    data: Dict[str, Any]
    created_at: datetime
    updated_at: datetime
